jerry-core/parser/js/js-parser.c

```diff
@@ -3162,14 +3153,9 @@ parser_parse_script (const uint8_t *arg_list_p, /**< function argument list */
 
     ecma_free_value (col_str_val);
     ecma_free_value (line_str_val);
-    ecma_deref_ecma_string (err_str_p);
+    ecma_free_value (err_str_val);
 #else /* !ENABLED (JERRY_ERROR_MESSAGES) */
-#if !ENABLED (JERRY_ESNEXT)
-    if (parser_error.error != PARSER_ERR_INVALID_REGEXP)
-#endif /* !ENABLED (JERRY_ESNEXT) */
-    {
-      ecma_raise_syntax_error ("");
-    }
+    ecma_raise_syntax_error ("");
```

Comment:
So the whole syntax error creation part must be kept, just the profile guards should be eliminated.